pkg/kubelet/disk_manager.go

```diff
@@ -83,15 +83,19 @@ func (dm *realDiskSpaceManager) getFsInfo(fsType string, f func() (cadvisorapi.F
 	return fsi, nil
 }
 
-func (dm *realDiskSpaceManager) IsDockerDiskSpaceAvailable() (bool, error) {
-	return dm.isSpaceAvailable("docker", dm.policy.DockerFreeDiskMB, dm.cadvisor.DockerImagesFsInfo)
+func (dm *realDiskSpaceManager) IsRuntimeDiskSpaceAvailable(runtime string) (bool, error) {
+	return dm.isSpaceAvailable(runtime, dm.policy.DockerFreeDiskMB, dm.cadvisor.ImagesFsInfo)
 }
 
 func (dm *realDiskSpaceManager) IsRootDiskSpaceAvailable() (bool, error) {
-	return dm.isSpaceAvailable("root", dm.policy.RootFreeDiskMB, dm.cadvisor.RootFsInfo)
+	return dm.isSpaceAvailable("root", dm.policy.RootFreeDiskMB, dm.rootFsWrapper)
 }
 
-func (dm *realDiskSpaceManager) isSpaceAvailable(fsType string, threshold int, f func() (cadvisorapi.FsInfo, error)) (bool, error) {
+func (dm *realDiskSpaceManager) rootFsWrapper(_ string) (cadvisorapi.FsInfo, error) {
```

Comment:
Another way of getting rid of this, and also keeping the old function signature of `isSpaceAvailable()` is the add the runtime type in  
```go
type cadvisorClient struct {
        runtimeType string
 	manager.Manager
 }
```